src/profile-logic/profile-data.js

```diff
@@ -2577,12 +2577,20 @@ export function extractProfileFilterPageData(
     return null;
   }
 
-  // Getting the first page's innerWindowID and then getting its url.
-  const innerWindowID = [...relevantPages][0];
-  const filteredPages = pages.filter(
-    page => page.innerWindowID === innerWindowID
+  // Getting the pages that are relevant and a topmost frame.
+  let filteredPages = pages.filter(
+    page =>
+      relevantPages.has(page.innerWindowID) && page.embedderInnerWindowID === 0
   );
 
+  if (filteredPages.length > 1) {
+    // If there are more than one topmost page, it's also good to filter out the
+    // `about:` pages so user can see their url they are actually profiling.
```

Comment:
It's not clear to me what happens here. Is it happening when navigating (for example between about:home and the next URL?)